### PR TITLE
Make AK/SK configurable for `obs_bucket` and `obs_bucket_object`

### LINKED ISSUE #713: OBS bucket creation not possible with agency auth
# Terraform Version
```
Terraform v0.13.5
+ provider registry.terraform.io/hashicorp/random v3.0.0
+ provider registry.terraform.io/opentelekomcloud/opentelekomcloud v1.21.4
```

# Affected Resource(s)
* opentelekomcloud_obs_bucket

# Terraform Configuration Files
```hcl
provider "opentelekomcloud" {
  agency_name        = "myagency"
  agency_domain_name = "OTC-EU-DE-00000000000000000000"
  delegated_project  = "eu-de"
}

resource "random_string" "random" {
  length = 16
  special = false
}

resource "opentelekomcloud_obs_bucket" "mybucket" {
  bucket = random_string.random.result
  acl    = "private"
}

```

# Steps to Reproduce
```
export OS_AUTH_URL=https://iam.eu-de.otc.t-systems.com/v3
export OS_AUTH_TOKEN=<mytoken>
terraform apply
```

# Expected Behavior

Bucket should be created

# Actual Behavior

Bucket creation fails

```
Error: Error creating OpenTelekomCloud OBS client: missing credentials for OBS, need access_key and secret_key or security_token values for provider

  on providers.tf line 12, in resource "opentelekomcloud_obs_bucket" "mybucket":
  12: resource "opentelekomcloud_obs_bucket" "mybucket" {
```

# Important Factoids

I know that the OBS bucket creation requires aksk values, but how do I proceed when using the agency auth. I use multiple providers, it is not possible for me to set the aksk token before.

### PR DESCRIPTION
**Pull request by outcatcher**
## Summary of the Pull Request
Resolves part of #713

## PR Checklist

* [x] Refers to: #713
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

Bucket:
```
=== RUN   TestAccObsBucket_basic
--- PASS: TestAccObsBucket_basic (40.68s)
=== RUN   TestAccObsBucket_aksk
--- PASS: TestAccObsBucket_aksk (20.74s)
=== RUN   TestAccObsBucket_tags
--- PASS: TestAccObsBucket_tags (22.41s)
=== RUN   TestAccObsBucket_versioning
--- PASS: TestAccObsBucket_versioning (38.09s)
=== RUN   TestAccObsBucket_logging
--- PASS: TestAccObsBucket_logging (24.17s)
=== RUN   TestAccObsBucket_lifecycle
--- PASS: TestAccObsBucket_lifecycle (20.83s)
=== RUN   TestAccObsBucket_website
--- PASS: TestAccObsBucket_website (20.73s)
=== RUN   TestAccObsBucket_cors
--- PASS: TestAccObsBucket_cors (21.30s)
PASS

Process finished with exit code 0
```

Object:
```
=== RUN   TestAccObsBucketObject_source
--- PASS: TestAccObsBucketObject_source (41.69s)
=== RUN   TestAccObsBucketObject_content
--- PASS: TestAccObsBucketObject_content (23.20s)
=== RUN   TestAccObsBucketObject_contentAKSK
--- PASS: TestAccObsBucketObject_contentAKSK (22.46s)
PASS

Process finished with exit code 0
```

### CODE REVIEW

**Comment by outcatcher** [comment]
recheck

File: opentelekomcloud/resource_opentelekomcloud_obs_bucket.go [hunk 1]
```diff
@@ -248,15 +251,71 @@ func resourceObsBucket() *schema.Resource {
...
+	if err != nil {
+		return nil, fmt.Errorf(errMsg, err)
+	}
+	return client, nil
+}
+
 func resourceObsBucketCreate(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	obsClient, err := config.newObjectStorageClient(GetRegion(d, config))
+	client, err := obsClient(d, config)
```
Comment (lego963) [inline opentelekomcloud/resource_opentelekomcloud_obs_bucket.go:316]: now it logs in obs, but it could be surprise for a user

**Review by lego963 — CHANGES_REQUESTED** [review]

File: opentelekomcloud/resource_opentelekomcloud_obs_bucket_object.go [hunk 2]
```diff
@@ -247,21 +271,27 @@ func resourceObsBucketObjectRead(d *schema.ResourceData, meta interface{}) error
...
 	return nil
 }
 
 func resourceObsBucketObjectDelete(d *schema.ResourceData, meta interface{}) error {
 	config := meta.(*Config)
-	obsClient, err := config.newObjectStorageClient(GetRegion(d, config))
+	client, err := obsClient(d, config)
 	if err != nil {
-		return fmt.Errorf("Error creating OpenTelekomCloud OBS client: %s", err)
+		return err
```
Comment (anton-sidelnikov) [inline opentelekomcloud/resource_opentelekomcloud_obs_bucket_object.go:294]: why you err without msg?

Comment (outcatcher) [reply opentelekomcloud/resource_opentelekomcloud_obs_bucket_object.go:294]: Error message is specified in `obsClient` function: `errMsg := "error creating OpenTelekomCloud OBS client: %s"`

File: opentelekomcloud/resource_opentelekomcloud_obs_bucket.go (hunk 1 above)
Comment (outcatcher) [reply opentelekomcloud/resource_opentelekomcloud_obs_bucket.go:316]: `obsClient` wraps `config.newObjectStorageClient(GetRegion(d, config))`. When using config without `credentials`, behaviour is not changing.

**Review by anton-sidelnikov — DISMISSED** [review]

**Review by lego963 — DISMISSED** [review]

**Review by anton-sidelnikov — APPROVED** [review]

**Review by lego963 — APPROVED** [review]

**Pull request merged**